app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
+  validates :title, presence: true
+  validates :text, presence: true
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- **[OPTIONAL]** Making this method a private method and then attaching an `after_save` callback to it would be good, so you don't have to call it yourself and anytime a post is saved, the method is triggered.

  ```
  after_save :update_posts_counter

  ...

  private

  def update_posts_counter
    author.update(posts_counter: author.posts.count)
  end
  ```